BaseTools/Scripts/GitHub.py

```diff
@@ -0,0 +1,160 @@
+## @file
+#  GitHub API helper functions.
+#
+#  Copyright (c) Microsoft Corporation.
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import logging
+import re
+import requests
+
+from collections import OrderedDict
+from edk2toollib.utility_functions import RunCmd
+from io import StringIO
+from typing import List
+
+"""GitHub API helper functions."""
+
+
+def leave_pr_comment(
+    token: str, owner: str, repo: str, pr_number: str, comment_body: str
+):
+    """Leaves a comment on a PR.
+
+    Args:
+        token (str): The GitHub token to use for authentication.
+        owner (str): The GitHub owner (organization) name.
+        repo (str): The GitHub repository name (e.g. 'edk2').
+        pr_number (str): The pull request number.
+        comment_body (str): The comment text. Markdown is supported.
+    """
+    url = f"https://api.github.com/repos/{owner}/{repo}/issues/{pr_number}/comments"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+    data = {"body": comment_body}
+    response = requests.post(url, json=data, headers=headers)
+    response.raise_for_status()
+
+
+def get_reviewers_for_current_branch(
+    workspace_path: str, maintainer_file_path: str
+) -> List[str]:
+    """Get the reviewers for the current branch.
+
+    Args:
+        workspace_path (str): The workspace path.
+        maintainer_file_path (str): The maintainer file path.
+
+    Returns:
+        List[str]: A list of GitHub usernames.
+    """
+
+    out_stream_buffer = StringIO()
+    cmd_ret = RunCmd(

```

Comment:
would it be better to use RunPythonScript: https://github.com/tianocore/edk2-pytool-library/blob/b9856f3d15c07c05007f44203335e8f6d8f96235/edk2toollib/utility_functions.py#L216

Reply:
Yes, updated.